presto/eval.py

```diff
@@ -143,7 +143,8 @@ def finetune_sklearn_model(
                 random_state=self.seed,
             ),
             "CatBoostClassifier": CatBoostClassifier(
-                random_state=self.seed, class_weights=class_weight_dict
+                random_state=self.seed,
+                class_weights={"True": class_weight_dict[1], "False": class_weight_dict[0]},
```

Comment:
I did not know `targets` was treated as a Boolean in CatBoost. It looks a bit weird to me. This is different behavior from `sklearn` models?

Reply:
Yes, it's different as far as I can tell. Its weird to me too.

Reply:
I think it's probably because CatBoost handles Boolean type natively while sklearn models will convert to `[0,1]`.
The core reason seems to orginate from [here](https://github.com/WorldCereal/presto-worldcereal/blob/c2d0964287a2f1cf9a42d876da4c6e249bd5ac76/presto/dataset.py#L61) and [here](https://github.com/WorldCereal/presto-worldcereal/blob/c2d0964287a2f1cf9a42d876da4c6e249bd5ac76/presto/dataset.py#L185) which returns Boolean arrays. I think it would be better to fix it here explicitly cast to `Int`. That way, the weird fix for the class weights here can also be reversed and we're back at consistency. WDYT?